serverless/serverless
# Install command

## What did you implement:

```
serverless install --url https://github.com/ac360/test-service
```

* Added a minimal `sls install` command to download a serverless service zip file from a Github repository via its URL, and unpack it.

* Accepts only a `--url` option.

* If a folder with the same name already exists in the current working directory, this will NOT overwrite it.

## How did you implement it:

Created a new `install` plugin in core.

## How can we verify it:

* Run this: `serverless install --url https://github.com/ac360/test-service`

* Check your current working directory to see if a `test-service` folder exists, with all of the service's files in it.

**Note:** Tests for downloading the actual service are not currently implemented since I'm not sure how doing local file operations works in Travis CI.

**Note #2:** This PR introduces a new npm dependency [download](https://github.com/kevva/download).  To get Travis CI to work, I had to rerun `npm shrinkwrap --dev`.  This hadn't been run for a while, so please be advised this was done in this PR.

## Todos:

- [x] Make sure code coverage hasn't dropped - It has, but tests may be tricky to implement.
